src/net.h

```diff
@@ -377,6 +377,8 @@ class CNode
     /** Offset inside the first vSendMsg already sent */
     size_t nSendOffset GUARDED_BY(cs_vSend){0};
     uint64_t nSendBytes GUARDED_BY(cs_vSend){0};
+    /** Disconnect after this or more bytes have been written to the socket. */
+    std::optional<uint64_t> m_disconnect_after_send_bytes GUARDED_BY(cs_vSend);
```

Comment:
Rather than doing this, it might make more sense to have simplified relay logic be to send the messages:

 * VERSION
 * [after receiving VERSION] VERACK
 * [after receiving VERACK] INV
 * [after receiving GETDATA] TX
 * PING
 * [after receiving PONG] disconnect

?

Reply:
Yes, and maybe also send `wtxidrelay`. Any reason not to send `TX` directly and avoid the `INV`, like below?

* VERSION
* [after receiving VERSION] WTXIDRELAY VERACK
* [after receiving VERACK] TX
* PING
* [after receiving PONG] disconnect

Reply:
> Any reason not to send `TX` directly and avoid the `INV`, like below?

It has been suggested to stop processing unrequested TXs messages, see #21224. While that didn't make it, the idea is still floating around. If bitcoin core itself started relying on unrequested msgs being processed, that would probably kill it once for all. 
Maybe that's ok, but we should make sure that there aren't alternative clients that would ignore unrequested TXs.

Reply:
Also, if you send the `INV` first and wait for a request to send the actual tx, you might be a little more confident that the peer you connected to is really relaying txs, not just gathering statistics.

I was thinking that you could send a fixed `VERSION` string and no feature messages (`WTXIDRELAY`, `SENDADDRV2`, etc), as a way of minimising how identifiable your node is versus other clients doing sensitive-relay. (I don't think you can have a larger anonymity set than that, since "send a single tx and close the connection" seems pretty distinct)

Reply:
We don't accept transactions while in IBD, so sending an INV first might be an easy way to catch that condition too. Checking that we're on the same tip, and respecting feefilter might also catch those cases, and be worth doing anyway though.

Reply:
> we should make sure that there aren't alternative clients that would ignore unrequested TXs

There is a precedent: from https://developer.bitcoin.org/reference/p2p_networking.html#tx

> [BitcoinJ](http://bitcoinj.github.io/) will send a [“tx” message](https://developer.bitcoin.org/reference/p2p_networking.html#tx) unsolicited for transactions it originates

Reply:
For now I kept it to send `TX` right away as this is simpler and somebody blackhole-ing the transaction is ok.